src/main/java/org/minions/devfund/sergio/WhackAMole.java

```diff
@@ -0,0 +1,174 @@
+package org.minions.devfund.sergio;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Main methods for game Whack a Mole.
+ *
+ * @author Sergio Navarro
+ * @since 4/12/2018
+ */
+public class WhackAMole {
+
+    private static final char GRID_WHITE_SPACE = '*';
+    private static final char MOLE = 'M';
+    private int molesLeft;
+    private int score;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+    private int gridDimension;
+
+    /**
+     * Constructor for WhackAMole.
+     *
+     * @param numAttempts   - The number of attempts for user.
+     * @param gridDimension - The dimension of the grid.
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        this.gridDimension = gridDimension;
+        moleGrid = new char[this.gridDimension][this.gridDimension];
+        loadGrid();
+    }
+
+    /**
+     * Return the current status of given position.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY -Represent the Y coordinate.
+     * @return The current status.
+     */
+    char getStatusPosition(int posX, int posY) {
+        return moleGrid[posX][posY];
+    }
+
+    /**
+     * Gets the score.
+     *
+     * @return The score.
+     */
+    int getScore() {
+        return score;
+    }
+
+    /**
+     * Gets the number of current attempts.
+     *
+     * @return The number of attempts.
+     */
+    int getAttemptsLeft() {
+        return attemptsLeft;
+    }
+
+    /**
+     * Places a Mole.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY - Represent the Y coordinate.
+     * @return True if the position (x,y) it is now a MOLE, false otherwise.
+     */
+    boolean place(int posX, int posY) {
+        if (moleGrid[posX][posY] != MOLE) {
+            moleGrid[posX][posY] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Whacks the grid attempting to hit a mole.
+     *
+     * @param posX - Represent the X coordinate.
+     * @param posY - Represent the Y coordinate.
+     */
+    void whack(int posX, int posY) {
+        char whack = 'W';
+
+        if (moleGrid[posX][posY] == MOLE) {
+            moleGrid[posX][posY] = whack;
+            molesLeft--;
+            score++;
+        }
+        attemptsLeft--;
+        printGridToUser();
+    }
+
+    /**
+     * Prints the current grid for user.
+     */
+    private void printGridToUser() {
```

Comment:
Now this method is no longer necessary. it was only for edx.
you can delete it !